modules/graph/tools/fragment_loader_impl.h

```diff
@@ -225,13 +295,24 @@ void dump_graph(Client& client, grape::CommSpec& comm_spec,
   auto const& fragments = fg->Fragments();
   for (const auto& item : fg->FragmentLocations()) {
     if (item.second == client.instance_id()) {
-      std::shared_ptr<fragment_t> fragment;
-      VINEYARD_CHECK_OK(client.GetObject(fragments.at(item.first), fragment));
-      if (options.dump_dry_run_rounds) {
-        traverse_fragment(fragment, options.dump_dry_run_rounds);
+      if (options.compact_edges) {
+        std::shared_ptr<fragment_compact_t> fragment;
+        VINEYARD_CHECK_OK(client.GetObject(fragments.at(item.first), fragment));
+        if (options.dump_dry_run_rounds) {
+          traverse_fragment(fragment, options.dump_dry_run_rounds);
+        } else {
+          dump_fragment(fragment, target_directory + "/output_graph_f" +
+                                      std::to_string(fragment->fid()));
+        }
       } else {
-        dump_fragment(fragment, target_directory + "/output_graph_f" +
-                                    std::to_string(fragment->fid()));
+        std::shared_ptr<fragment_t> fragment;
+        VINEYARD_CHECK_OK(client.GetObject(fragments.at(item.first), fragment));
+        if (options.dump_dry_run_rounds) {
+          traverse_fragment(fragment, options.dump_dry_run_rounds);
+        } else {
+          dump_fragment(fragment, target_directory + "/output_graph_f" +
+                                      std::to_string(fragment->fid()));
```

Comment:
Try add the fragment_t as the template type parameter for `traverse_fragment` and `dump_fragment` to avoid the duplication.